src/state/slices/txHistorySlice/utils.ts

```diff
@@ -41,7 +41,7 @@ export const serializeTxIndex = (
   data?: Tx['data'],
 ): TxIndex => {
   // special case for thorchain transactions sent back in multiple parts
-  if (data && data.parser === 'swap') {
+  if (data && data.parser === 'swap' && !address.toLowerCase().startsWith('osmo')) {
```

Comment:
Rather than explicitly excluding `osmo` (something we'd need to do any time a something else used `swap` in their `parser` key), we'll instead want to narrow it to ensure that it's only acting on THORSwaps.

The below should do the trick as a more future proof approach:

```suggestion
  if (data && data.parser === 'swap' && address.toLowerCase().startsWith('thor')) {
```